### PR TITLE
feat: add basic_auth and tls in etcd config

### PR DESCRIPTION
**Pull request by rfyiamcool**
### change

add basic_auth and tls in etcd config.

### CODE REVIEW

**Comment by git-hulk** [comment]
@rfyiamcool wow, thanks for your quick PR.

**Comment by rfyiamcool** [comment]
> @rfyiamcool wow, thanks for your quick PR.

😁

File: storage/persistence/etcd/etcd.go [hunk 1]
```diff
@@ -26,21 +26,37 @@ import (
...
 	defaultDailTimeout = 5 * time.Second
 )
 
+type Config struct {
+	Addrs     []string `yaml:"addrs"`
+	BasicAuth struct {
+		Enable   bool   `yaml:"enable"`
+		Username string `yaml:"username"`
+		Password string `yaml:"password"`
+	} `yaml:"basic_auth"`
```
Comment (git-hulk) [inline storage/persistence/etcd/etcd.go]:
```suggestion
    Username string `yaml:"username"`
	Password string `yaml:"password"`
```

The etcd client will use the auth configuration only if the username and password are not empty: https://github.com/etcd-io/etcd/blob/main/client/v3/client.go#L388-L392. So I think we can assign the username/password to etcd.Client directly.

**Comment by git-hulk** [comment]
Others are good to me, great thanks for your contributions.

File: storage/persistence/etcd/etcd.go (hunk 1 above)
Comment (rfyiamcool) [reply storage/persistence/etcd/etcd.go]: done 😁

**Review by git-hulk — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]

**Pull request merged**